### PR TITLE
SADL web prototype - Xtext 2.11.RC1

### PR DESCRIPTION
**Pull request by kittaakos**
Requirements:
 - Java 8
 - Xtext 2.11.RC1
 - Apache Maven 3.3.9 (More details are available [here].)

[here]: https://github.com/crapo/sadlos2/tree/xtext_web_prototype/sadl3/deployment

```
git clone https://github.com/crapo/sadlos2.git \
&& cd sadlos2 \
&& git checkout origin/xtext_web_prototype -b xtext_web_prototype \
&& cd sadl3/com.ge.research.sadl.parent/ \
&& mvn clean verify \
&& ./gradlew build
```

Install the tooling features of Xtext 2.11.RC1 version into the Eclipse IDE.
`Install New Softwares...` > Add the followings link http://download.eclipse.org/modeling/tmf/xtext/updates/milestones/head/S201701180206/ and select Xtext, LSP4J, and Uncategorized. After the installation, restart the Eclipse IDE and install the projects (`sadl3/com.ge.research.sadl.parent/` with the nested projects) into the workspace.
Install Buildship tooling support (for Gradle) into Eclipse IDE. Choose version 2.0.0.v20161108-1534-m or later from http://download.eclipse.org/buildship/updates/e45/milestones/2.x/.
Open target platform file under `com.ge.research.sadl.target`, wait until the target definition gets resolved, then set the resolved definition as the target platform.

If errors still exist in any of the following projects; `com.ge.research.sadl.ide`, `io.typefox.lsp.endpoint`, `io.typefox.lsp.monaco`, then just select thos projects and from the context menu execute `Gradle` > `Refresh Gradle Project`.
Or just clean the projects from `Project` > `Clean...`.

Note:
 - The `com.ge.reasearch.sadl.errorgenerator` project has been renamed to `com.ge.research.sadl.errorgenerator`.
 - The SADL model and the inferencer processors are available in the headless case as well. The SPI is used for the service discovery.

### CODE REVIEW

File: sadl3/com.ge.research.sadl.parent/com.ge.reasearch.sadl.errorgenerator/.classpath [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kittaakos) [inline sadl3/com.ge.research.sadl.parent/com.ge.reasearch.sadl.errorgenerator/.classpath]: This changes and the two below as well is from the `reaserach` project which we have removed and replaced with `the c.g.research.sadl.errorgenerator`.
Please leave as is, we will get rid of this when we have merged to master. Just wanted to clarify the commit comment. Also, please note, Eclipse will complain when you try to import both the `research` and `reaserch` projects into workspace.

**Comment by crapo** [comment]
Akos,

I was sorry to hear you were ill. Are you working today? If so, can we have a webex/tcon to get this merge complete.

Regards,
Andy

File: sadl3/com.ge.research.sadl.parent/com.ge.reasearch.sadl.errorgenerator/src/sadlMessages.properties [hunk 2]
```diff
@@ -20,15 +20,15 @@ atpt_no_selection.description=When running ATPT, you must select the right item
 atpt_no_xml_selected=No folder or xml file selected for ATPT
 atpt_no_xml_selected.description=To run ATPT from eclipse, you must select the associated common test procedure (CTP) \
 									output file or folder which contains the CTP output file(s).
-cannot_assign_existing=Cannot assign {0}: property '{1}' already has {0} assigned to '{2}'.
+cannot_assign_existing=Cannot assign {0}: property ''{1}'' already has {0} assigned to ''{2}''.
```
Comment (kittaakos) [inline sadl3/com.ge.research.sadl.parent/com.ge.reasearch.sadl.errorgenerator/src/sadlMessages.properties]: We should apply these changes on the `com.ge.research.sadl.errorgenerator` project and not on the `com.ge.reasearch.sadl.errorgenerator`. I will make those changes and make sure, the project, with the typo, is gone.

Please note, there is a typo in the latter one.

**Comment by kittaakos** [comment]
<b>NOTE:</b> reverted via https://github.com/crapo/sadlos2/pull/152.

**Pull request merged**